firedrake/mesh.py

```diff
@@ -3759,24 +3741,6 @@ def _parent_mesh_embedding(
     input_ranks = np.compress(locally_visible, input_ranks_global, axis=0)
     input_coords_idxs = np.compress(locally_visible, input_coords_idxs_global, axis=0)
 
-    if not exclude_halos and parent_mesh.comm.size > 1:
-        # Reorder the halos so that off rank points are at the end of the local
-        # ordering. This is required for dat views to work correctly.
-        owned_ranks_tosort = owned_ranks.copy()
-        # seting all off rank points to comm.size will ensure they are at the
-        # end when we sort by rank (but before any points that are not found on
-        # any rank which will have been set to comm.size + 1)
-        owned_ranks_tosort[owned_ranks != parent_mesh.comm.rank] = parent_mesh.comm.size
-        # now a sort by rank will give us the ordering we want
-        idxs = np.argsort(owned_ranks_tosort)
-        coords_embedded = coords_embedded[idxs]
-        global_idxs = global_idxs[idxs]
-        reference_coords = reference_coords[idxs]
-        parent_cell_nums = parent_cell_nums[idxs]
-        owned_ranks = owned_ranks[idxs]
-        input_ranks = input_ranks[idxs]
-        input_coords_idxs = input_coords_idxs[idxs]
-
```

Comment:
What has replaced this?

Reply:
If it's `dmcommon.plex_renumbering`, how does setting up the `pointSF` of the swarm play nicely with this since you do it afterwards?

Reply:
Yes, we now use the existing renumbering infrastructure. `pointSF` is more fundamental. `pointSF` defines the association: `swarm point <- (rank, swarm point)`. `_renumbering` depends on `pointSF` and is used to define Firedrake cell numbers, DoF orderings, etc.

Reply:
The the local ordering of the `pointSF` has presumably changed then, since you don't move halos to the end?

Reply:
Yes. That is what the third argument of `_make_input_ordering_sf` is.